subprojects/core/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -673,4 +676,16 @@ public boolean isIgnoreExitValue() {
     public List<CommandLineArgumentProvider> getJvmArgumentProviders() {
         return javaExecHandleBuilder.getJvmArgumentProviders();
     }
+
+    /**
+     * Returns the result for the command run by this task. Returns {@code null} if this task has not been executed yet.
+     *
+     * @return The result. Returns {@code null} if this task has not been executed yet.
+     * @since 6.1
+     */
+    @Internal
+    @Incubating
+    public ExecResult getExecResult() {
```

Comment:
Sorry to jump in late, but I think this would be much better represented as a `Provider<ExecResult>`, so that people can easily wire this result in as an input to other tasks, rather than as method that sometimes results `null` and sometimes returns a result.

The provider should fail when queried before the task has completed. And later, connecting this provider to an `@Input` of another task would imply a dependency on the `JavaExec` task (we can do this as improvement in a later PR).

Reply:
Great feedback and sorry for missing this.

@runningcode Could you look at making this change to the PR?

Reply:
Hey, thanks for the feedback on the PR. I will have some time to address both these points in two weeks because of my current work and travel schedule.